packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -2567,30 +2591,38 @@ class BuildOwner {
               ),
               DiagnosticsProperty<Element>(
                 'The offending element (which does not appear to be a descendant of the root of the build scope) was',
-                _dirtyElements[index],
+                element,
                 style: DiagnosticsTreeStyle.errorProperty,
               ),
             ]);
           }
           return true;
         }());
+        if (!kReleaseMode && debugProfileBuildsEnabled) {
+          Timeline.startSync(
+            '${element.widget.runtimeType}',
+            arguments: element.widget.toDiagnosticsNode().toTimelineArguments(),
```

Comment:
Is there a performance cost associated with including all this information? Will this make it harder to compare relative performance of two widget builds if one widget includes a lot of information and the other doesn't?

Reply:
I guess, the performance overhead of this is "charged" to the overall "BUILD" task and not the individual widgets, so it shouldn't make those comparisons harder?

Reply:
It's charged to the parent widget.

My perspective was that we'd mostly given up on making these traces representative of actual time. The tracing alone dramatically skews the numbers. It does go rather against https://github.com/flutter/flutter/issues/93262 though.

Reply:
Can we put these extra arguments behind a flag then? While performance debugging customer:money it has been helpful to at least see relative performance differences between the individual widgets in the `debugProfileBuildsEnabled` mode to identify "slow widgets" (same for RenderObjects). As I said in #93262 I'd rather want make this mode faster than slower to make performance debugging even easier. Handling two traces (one with the extra profiling enabled and one without to get accurate numbers) has not been practical.

Reply:
There's already a flag: `--profile`. The arguments aren't added in profile builds. (In debug builds the traces aren't representative anyway because of JIT/AOT differences.)

Reply:
oh that said i see that i missed the case that you're commenting on, oops

Reply:
ah, it looks like i missed this in _all_ the important cases. Somehow I remembered to protect it only in the rare cases.